app/js/components/manageApps/ManageApps.jsx

```diff
@@ -143,64 +147,103 @@ export default class ManageApps extends React.Component {
   }
 
   handleUpload() {
-    const zipedAddon = this.state.files[0];
-    const readZippedAddon = new Promise((resolve, reject) => {
-      let new_zip = new JSZip();
-      new_zip.loadAsync(zipedAddon)
-        .then((zip) => {
-          return zip.files['manifest.webapp'] ? zip.files['manifest.webapp'] : false;
-        })
-        .then((file) => {
-          return file ? new_zip.file(file.name).async("string") : false;
-        })
-        .then(result => {
-          resolve(JSON.parse(result));
-        })
-        .catch(e => {
-          resolve(false);
-        });
-    });
+    const uploadFile = this.state.files[0];
+    if (uploadFile.type === 'application/zip') {
+      const readZippedAddon = new Promise((resolve, reject) => {
+        let new_zip = new JSZip();
+        new_zip.loadAsync(uploadFile)
+          .then((zip) => {
+            return zip.files['manifest.webapp'] ? zip.files['manifest.webapp'] : false;
+          })
+          .then((file) => {
+            return file ? new_zip.file(file.name).async("string") : false;
+          })
+          .then(result => {
+            resolve(JSON.parse(result));
+          })
+          .catch(e => {
```

Comment:
I would recommend you use better variable naming eg error / exception instead of 'e'